src/app/auth/login.ts

```diff
@@ -113,6 +132,331 @@ export const loginWithPhoneCookie = async ({
+  let hasPhoneAccount = true
+  if (phoneAccount instanceof CouldNotFindUserFromPhoneError) {
+    hasPhoneAccount = false
+  } else {
+    if (phoneAccount instanceof Error) return phoneAccount
+  }
+  if (hasPhoneAccount) {
+    // is there still txns left over on the device account?
+    const deviceWallets = await WalletsRepository().listByAccountId(account.id)
+    if (deviceWallets instanceof Error) return deviceWallets
+    const ledger = LedgerService()
+    let deviceAccountHasBalance = false
+    for (const wallet of deviceWallets) {
+      const balance = await ledger.getWalletBalance(wallet.id)
+      if (balance instanceof Error) return balance
+      if (balance > 0) {
+        deviceAccountHasBalance = true
+      }
+    }
+    if (deviceAccountHasBalance) {
+      /// Scenario 1.5 - has txns on device account but phone account exists
+      return new PhoneAccountAlreadyExistsNeedToSweepFunds(
+        `Error phone account already exists. You need to manually sweep funds
+        to your phone account. Here are the steps to do this:
+        1. login with phone account and create an invoice (save this invoice)
+        2. logout of the phone account
+        3. login with device account (click "Explore Wallet Instead" button)
+        4  Pay the invoice with max wallet amount.
+        5. Logout of the device account
+        6. Log into the phone account and check for the funds`,
+      )
+    } else {
+      /// Scenario 1 - no txns on device account but phone account exists
+      return new PhoneAccountAlreadyExists(
+        "Phone Account already exists. Please logout and log back in with your phone account.",
+      )
+    }
+  }
+
+  // Scenario 2 - Happy Path
+  // a. update kratos
+  const kratosAccount = await AccountsRepository().findByUserId(account.kratosUserId)
+  if (kratosAccount instanceof Error) return kratosAccount
+  const updatedKratosAccount = await AuthWithDeviceAccountService().upgradeToPhoneSchema({
+    kratosUserId: account.kratosUserId,
+    phone,
+  })
+  if (updatedKratosAccount instanceof Error) return new Error("updatedKratosAccount")
+  // b. update user
+  const userRemovedDevice = await UsersRepository().unsetDeviceIdForUser(
+    account.kratosUserId,
+  )
+  if (userRemovedDevice instanceof Error) return userRemovedDevice
+  const userAddedPhone = await UsersRepository().update({
+    id: account.kratosUserId,
+    phone,
+  })
+  if (userAddedPhone instanceof Error) return userAddedPhone
+  // c. update account
+  kratosAccount.level = AccountLevel.One
+  const accountLevelOne = await AccountsRepository().update(kratosAccount)
+  if (accountLevelOne instanceof Error) return accountLevelOne
+
+  // returning a new session token
+  const authPhone = AuthWithPhonePasswordlessService()
+  const kratosResult = await authPhone.loginToken({ phone })
+  if (kratosResult instanceof Error) return kratosResult
+  return kratosResult.sessionToken
+}
+
+export const loginWithDevice = async ({
```

Comment:
it is meant to only work when the deviceAlready exist? 

how one know when to go here, versus https://github.com/GaloyMoney/galoy/pull/2071/files#diff-4b614afac7a950ff856e0a8f9ed1c3b92c23281e1e1ebe3b297d983b77b5b280R8 ?

Reply:
it's meant to follow the flow we already have for the phone login. If the device account already exists it logs you in, otherwise it creates the account.

i just removed the legacy device flow in this commit https://github.com/GaloyMoney/galoy/pull/2071/commits/57df85ec4364d20a3d8141ec3e6f42c74c86ded5